jwst/extract_1d/apply_apcorr.py

```diff
@@ -0,0 +1,271 @@
+import abc
+
+from typing import Tuple, Union, Type
+from scipy.interpolate import interp2d, interp1d
+from astropy.io import fits
+
+from ..assign_wcs.util import compute_scale
+from ..datamodels import DataModel
+
+
+class ApCorrBase(abc.ABC):
+    """Base class for aperture correction classes.
+
+    Parameters
+    ----------
+    input_model : `~/jwst.datamodels.DataModel`
+        Input data model used to determine matching parameters.
+    apcorr_table : `~/astropy.io.fits.FITS_rec`
+        Aperture correction table data from APCORR reference file.
+    location : tuple, Optional
+        Reference location (RA, DEC) used to calculate the pixel scale used in converting values in arcec to pixels.
+        Default is None, however, if the input reference data contains size/radius data in units of arcsecs, a location
+        is required.
+    apcorr_row_units : str, Optional
+        Units for the aperture correction data "row" values (assuming the aperture correction is a 2D array).
+        If not given, units will be determined from the input apcorr_table ColDefs if possible.
+
+    Raises
+    ------
+    ValueError :
+        If apcorr_row_units are not supplied and units are undifined in apcorr_table ColDefs
+    ValueError :
+        If the input apcorr_table cannot be reduced to a single row based on match criteria from input_model.
+
+    """
+    match_pars = {
+        'MIRI': {
+            'LRS': {'subarray': ['name']},
+            'MRS': {'instrument': []},  # Only one row is available for this mode; no selection criteria
+        },
+        'NIRSPEC': {
+            'IFU': {'instrument': ['filter', 'grating']},
+            'MSASPEC': {'instrument': ['filter', 'grating']},
+            'FIXEDSLIT': {'instrument': ['filter', 'grating']},  # Slit is also required; passed in as init arg
+            'BRIGHTOBJ': {'instrument': ['filter', 'grating']}
+        },
+        'NIRCAM': {
+            'WFSS': {'instrument': ['filter', 'pupil']},
+            'NRC_GRISM': {'instrument': ['filter', 'pupil']}
```

Comment:
Ok, then it might be as simple as removing that. 

If I remember correctly, I tried to create the different branches based on possible `EXPTYPE`s in JWST-CRDS, but if it's not going to be used, then removing it should work just fine

Reply:
Your correct! The rmap definitely has issues of this exact nature (I forget which mode I was testing... might've even been this one). A CRDS issue was filed for it. 

@stscieisenhamer do you think that the CRDS issue for the APCORR rmap covers this? Or is it too specific

Reply:
Found the issue: [CRDS-355](https://jira.stsci.edu/browse/CRDS-355)